apollo-router/src/plugins/authentication/jwks.rs

```diff
@@ -199,33 +206,23 @@ pub(super) async fn get_jwks(url: Url) -> Option<JwkSet> {
             e
         })
         .ok()?;
+
+    // remove any keys that can't be parsed
     raw_json.get_mut("keys").and_then(|keys| {
         keys.as_array_mut().map(|array| {
-            array.retain(|key| {
-                let alg_json = key.get("alg").unwrap_or_else(|| &Value::Null);
-                // Retention rules are, retain if:
-                //  - we can't find an alg field
-                //  - alg field isn't a string
-                //  - jsonwebtoken supports the algorithm (determined by creating an enum member)
-                if alg_json == &Value::Null || !alg_json.is_string() {
-                    true
-                } else {
-                    let alg_name = alg_json.as_str().expect("we checked it's a string");
-                    match Algorithm::from_str(alg_name)
```

Comment:
why remove the algorithm checking?

Reply:
It's checked by virtue of it trying to parse the individual key.

The error message can be seen in the tests.

Reply:
nice!